Sources/Interaction/Style/InteractorStyleManipulator/index.js

```diff
@@ -181,21 +181,21 @@ function vtkInteractorStyleManipulator(publicAPI, model) {
   //-------------------------------------------------------------------------
   publicAPI.removeMouseManipulator = (index) => {
```

Comment:
You should expect the arg to be the manipulator not the index since, all your capture index will be messed up by any remove.

Reply:
Why? That's a very normal API to be able to remove an element at an index that causes the content to shift:
* c++: [erase](http://www.cplusplus.com/reference/vector/vector/erase/)
* python: [pop](https://docs.python.org/3/tutorial/datastructures.html)
* js: [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
* swift: [remove](https://developer.apple.com/documentation/swift/array/1641390-remove)

I agree we could have an API to use the manipulator as an input, but shouldn't that be new methods? So we can keep those methods a bit more consistent in behavior and input?

Reply:
And how do you get the index of the manipulator that you want to remove? The index that was given to you at the registration is what became invalid after the first remove.

Reply:
pop / erase and so on make sens because you control the list, here you don't hence the issue.

Reply:
The user should not have to know that the internal representation is a list of some kind.

The thing that the user cares about is to be able to add/remove his interactor style. You should not add any extra burden to the dev to actually know how the internal of that class is made of. If so that leads to bad design.

before the design was:

```
id = manip.addStyle(style);
manip.removeStyle(id);
```

the design now should be

```
manip.addStyle(style);
manip.removeStyle(style);
```

Reply:
And we can maintain the previous api valid by returning `style` when calling `addStyle`, that way it is not a breaking change.

Reply:
So many comments 🙈

> And how do you get the index of the manipulator that you want to remove? The index that was given to you at the registration is what became invalid after the first remove.

Yeah indeed, that makes sense.

Reply:
What about `getManipulator(id)` ?

Reply:
No

Reply:
I want all the following to work

```
id = manip.addStyle(style);
manip.removeStyle(id);
```

```
manip.addStyle(style);
manip.removeStyle(style);
```